CriticalMass/URLCodable.swift

```diff
@@ -46,28 +46,34 @@ extension URLCodable {
         return try self.init(scheme: urlComponents.scheme, host: urlComponents.host, path: urlComponents.path, queryObject: codableObject)
     }
 
-    private func encode<T: Encodable>(codable: T) throws -> [URLQueryItem] {
+    private static func decode<T: Codable>(type: T.Type, from items: [URLQueryItem]) throws -> T {
+        let dict = items.reduce(into: [String: String]()) { result, item in
+            result[item.name] = item.value
+        }
+        let data = try JSONSerialization.data(withJSONObject: dict, options: [])
+        return try JSONDecoder().decode(type, from: data)
+    }
+}
+
+extension URLQueryItem {
+    static func encode<T: Encodable>(codable: T) throws -> [URLQueryItem] {
```

Comment:
Could also be renamed IMO

Reply:
Maybe then pull
```swift
return try dict.lazy.map { (key, value) -> URLQueryItem in
            switch value {
            case let stringValue as String:
                return URLQueryItem(name: key, value: stringValue)
            case let stringConvertible as CustomStringConvertible:
                return URLQueryItem(name: key, value: String(describing: stringConvertible))
            default:
                // More types are currently not supported and should be added if needed
                throw URLCodableError.encodingFailed
            }
        }.sorted { a, b in
            // The sorting is only required to make testing easier
            a.name > b.name
        }
```
out?

Reply:
I simplified the method a bit. Is it better now?